### ISSUE TITLE
bug: unable to execute custom schematic application blueprint

### ISSUE THREAD

**Issue by Hotell** [angular/devkit#528]
### Bug Report or Feature Request (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
```

### Area
```
- [ ] devkit
- [x] schematics
```

### Versions
Angular CLI: 1.7.3
Node: 8.10.0
OS: darwin x64

### Repro steps
```
npm i -g @martin_hotell/schematics typescript @angular/cli @angular-devkit/schematics-cli

ng new better-cli --material --collection @martin_hotell/schematics -d
```

### The log given by the failure
![image](https://user-images.githubusercontent.com/1223799/37570394-9adacfca-2aef-11e8-8093-7bf201facf82.png)

### Desired functionality
being able to generate new app via custom schematic

### Mention any other details that might be useful
There were issues created before which were closed for unknown reasons :) not cool at all 👀

**Comment by james-schwartzkopf** [comment]
Just a shot in the dark, you seem to be mixing npm and yarn.  Are you sure you installed your global schematics and global ng in the same one?  Could one of them be an older version or something?

**Comment by Hotell** [comment]
> just a shot in the dark, you seem to be mixing npm and yarn. Are you sure you installed your global schematics and global ng in the same one? Could one of them be an older version or something?

Nope, I'm not mixing anything, yarn is used only as task runner in local cli project in my bug issue + I'm explicitly described the reproduction steps.

**Comment by choucry13** [comment]
hey, any news ?

**Comment by Hotell** [comment]
@choucry13 my 50 cents are there will be no news in terms of fixing this in cli 1.x, just closed issue, cause it will be resolved with cli 6. Also CLI team has hands full with making everyting ready for 6, so that's why no one responded as well, other more important priorities in pipeline 👀✌️

**Comment by Brocco** [comment]
_NOTE: The following information is for version 1.4-1.7 (not 6+)_

To use a custom schematic for `ng new` you need to specify the collection (as you've done above), and you need to specify the schematic to use (which is by default "application").

If you'd like to use a different schematic name, you will need to configure that in your global `.angular-cli.json` file. The reference schema is [here](https://github.com/angular/angular-cli/blob/1.7.x/packages/@angular/cli/lib/config/schema.json#L619-L623).

This global value can also be set using...
```
ng set defaults.schematics.newApp yourSchematicName
```
This needs to be global because at the time that ng new is run there is no config file other than your global config file.

**Comment by Hotell** [comment]
I already know all of that, but what you're saying is I cannot execute application schematic from custom collection by providing it explicitly and I can only do that by setting the collection to global ng config ? That's sound like bug to me.

**Comment by Brocco** [comment]
I don't disagree; however, this is an issue with the CLI (not devkit) can you please create an issue on that [repo](https://github.com/angular/angular-cli/issues/new) with a link to this issue?

**Comment by alexeagle** [comment]
This issue was moved to angular/angular-cli#12131
